Standardize on the term "coordinator"

begriffs opened the pull request:
(except in code where it is still called master)

@ozgune I updated this based on your request from a few days ago

ozgune commented:
Hey @begriffs -- this is exciting!

I have three higher level questions. What do you and @sumedhpathak think about them?

(1) We have [an architectural diagram](https://docs.citusdata.com/en/v6.1/aboutcitus/introduction_to_citus.html) in our docs that use the terms "master node" and CitusDB. Should we update this diagram or open a separate issue for it?

(2) We have a few tags and links in our docs that use the term "master". For example, we have links to _master_node_failures, _production_deb_master_node, and _production_rhel_master_node. Now that we changed the titles, should we update these links too?

(3) In our Installation instructions and [User Defined Functions](https://docs.citusdata.com/en/v6.1/reference/user_defined_functions.html) reference, we provide steps that refer to "master". For example, Docker starts up a citus_master and Fedora gives instructions to create master directories. When we refer to the coordinator node in these sections, should we say "coordinator (master) node" to clarify the terminology to the user?

sumedhpathak commented:
> (1) We have an architectural diagram in our docs that use the terms "master node" and CitusDB. Should we update this diagram or open a separate issue for it?

Maybe let's update it now, but as a separate issue? @begriffs would it take long to update this diagram? We have this issue open: #22 for updating various other aspects of this diagram.

> (2) We have a few tags and links in our docs that use the term "master". For example, we have links to _master_node_failures, _production_deb_master_node, and _production_rhel_master_node. Now that we changed the titles, should we update these links too?

@begriffs I think we can update this in the current PR.

> (3) In our Installation instructions and User Defined Functions reference, we provide steps that refer to "master". For example, Docker starts up a citus_master and Fedora gives instructions to create master directories. When we refer to the coordinator node in these sections, should we say "coordinator (master) node" to clarify the terminology to the user?

@ozgune Which sections in the UDF section? We have UDF's which are named as master_xxx, which clearly can't change. For the Docker ones, is that an issue in the Docker repo instead of Docs?

begriffs commented:
I updated the diagram (just the minimal changes for this issue, not all things identified by #22). Also changed internal link names.

Names like citus_master come from the [docker-compose](https://github.com/citusdata/docker/blob/master/docker-compose.yml) file in the docker repo. But I do qualify in the docs, "master (aka coordinator)" before mentioning it. Also changed the master directory name in the single-node docs.

ozgune commented on installation/single_machine_docker.rst:
```diff
@@ -45,7 +45,7 @@ The first time you start the cluster it builds its containers. Subsequent startu
     Bind for 0.0.0.0:5432: unexpected error address already in use
 
-  This is because the "master" service attempts to bind to the standard PostgreSQL port 5432. Simply adjust :code:`docker-compose.yml`. Under the :code:`master` section change the host port from 5432 to 5433 or another non-conflicting number.
+  This is because the "master" (aka coordinator) service attempts to bind to the standard PostgreSQL port 5432. Simply adjust :code:`docker-compose.yml`. Under the :code:`master` section change the host port from 5432 to 5433 or another non-conflicting number.
```
Minor proposal.

(aka coordinator) -> (coordinator)

ozgune commented on installation/single_machine_docker.rst:
```diff
@@ -54,7 +54,7 @@ The first time you start the cluster it builds its containers. Subsequent startu
 **3. Verify that installation has succeeded**
 
-To verify that the installation has succeeded we check that the master node has picked up the desired worker configuration. First start the psql shell on the master node:
+To verify that the installation has succeeded we check that the coordinator node has picked up the desired worker configuration. First start the psql shell on the coordinator node:
```
Minor.

First start the psql shell on the coordinator (master) node:

ozgune commented on installation/single_machine_rhel.rst:
```diff
@@ -32,28 +32,28 @@ Let's create directories for those nodes to store their data. For convenience in
 
   cd ~
-  mkdir -p citus/master citus/worker1 citus/worker2
+  mkdir -p citus/coord citus/worker1 citus/worker2
```
Two quick questions.

1. Do we want to change this or leave it as-is?
2. If we change it, what do you think about "coord" --> "coordinator"?

ozgune commented on reference/sql_workarounds.rst:
```diff
@@ -127,7 +127,7 @@ Attempting to execute a JOIN between a local and a distributed table causes an e
   ERROR: cannot plan queries that include both regular and partitioned relations
 
-There is a workaround: you can replicate the local table to a single shard on every worker and push the join query down to the workers. We do this by defining the table as a 'reference' table using a different table creation API. Suppose we want to join tables *here* and *there*, where *there* is already distributed but *here* is on the master database.
+There is a workaround: you can replicate the local table to a single shard on every worker and push the join query down to the workers. We do this by defining the table as a 'reference' table using a different table creation API. Suppose we want to join tables *here* and *there*, where *there* is already distributed but *here* is on the coordinator database.
```
Completely unrelated small note. We have example tables named here and there?

begriffs replied:
Any proposals for new names? A and B; local_table and distributed_table? In the absence of a theme for this section, the names "here" and "there" seemed to work although they are a bit quirky.

ozgune commented on installation/single_machine_deb.rst:
```diff
@@ -33,28 +33,28 @@ Let's create directories for those nodes to store their data. For convenience in
 
   cd ~
-  mkdir -p citus/master citus/worker1 citus/worker2
+  mkdir -p citus/coord citus/worker1 citus/worker2
```
Depending on our answer above, could we also update this section accordingly?

ozgune commented:
@begriffs -- I quickly went over the changes added small comments. Once we agree on these comments, I could do a more detailed read over to make sure that we haven't missed anything -- or we could have another team member do that too. Exciting!

sumedhpathak commented on installation/single_machine_docker.rst:
```diff
@@ -54,7 +54,7 @@ The first time you start the cluster it builds its containers. Subsequent startu
+To verify that the installation has succeeded we check that the coordinator node has picked up the desired worker configuration. First start the psql shell on the coordinator (master) node:
 
 .. code-block:: bash
 
```
So I presume the below citus_master is hard-coded somewhere in Docker itself?

begriffs replied:
It's hard-coded in our docker compose file https://github.com/citusdata/docker/blob/master/docker-compose.yml#L5

sumedhpathak commented on reference/append.rst:
```diff
@@ -270,14 +270,14 @@ $$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$
 
 ::
 
     psql -h worker-node-1 -c "\COPY events FROM 'data.csv' WITH (FORMAT CSV, MASTER_HOST 'master-node')"
```
Could we replace the value part of this with coordinator-node-101 or something?
e.g.
`psql -h worker-node-1 -c "\COPY events FROM 'data.csv' WITH (FORMAT CSV, MASTER_HOST 'coordinator-node-101')"`

I noticed another example with the MASTER_HOST option, which we could change to coordinator. 
`\copy github_events from 'github_events-2015-01-01-0.csv' WITH (format CSV, master_host 'master-host-101')`

begriffs replied:
I think I'll make a few changes actually:

* call it worker-host-1 because it's being used as a host argument
* remove the -1 suffix because we don't mention a worker-host-2
* hence not add -101 to the coordinator since there can be only one

sumedhpathak commented on reference/append.rst:
```diff
@@ -207,8 +207,8 @@ Scaling Data Ingestion
+$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$$
 
 To ingest data into an append distributed table, you can use the `COPY <http://www.postgresql.org/docs/current/static/sql-copy.html>`_ command, which will create a new shard out of the data you ingest. COPY can break up files larger than the configured citus.shard_max_size into multiple shards. COPY for append distributed tables only opens connections for the new shards, which means it behaves a  ...
 
```
Below, could we update:
``` SQL

     -- In a separate transaction, append the staging table
     SELECT master_append_table_to_shard(select_events_shard(), 'stage_1', 'master-node', 5432);
```

and update the 'master-node' value with something like 'coordinator-node-101'?

sumedhpathak commented:
@begriffs Ping on this one. This one is also ready to go.

begriffs commented:
I was considering cherry-picking this to 6.1, but decided not to. It changes some internal link names like `_production_deb_master_node` -> `_production_deb_coordinator_node` and could conceivably break links going to the docs. That and the 6.2 version will soon replace 6.1 as the default.
